### PR TITLE
chore: switch to more proper webpack chunking

### PR DESCRIPTION
**Pull request by Artoria2e5**
Why fuck with eval when you can let webpack do it?

PS: the vendor split is deterministic for the entry-point list and the dependencies used by the entry points. WebpackHtmlPlugin could allegedly do that automatically, but guess what IT DOES NOT. I wonder how `index.html` gets it.

### CODE REVIEW

**Comment by Artoria2e5** [comment]
oops, the curse of the force-push...

**Comment by Jack-Works** [comment]
Cool. But I prefer the `WebpackHtmlPlugin` way.

Creating a new background page and let webpack inject chunks.

And further, I think we can create a "content script page" to get the dependencies and parse that page in the background page in runtime, we can get rid of thoes hacks on HTMLScriptElement.prototype.src

**Comment by Artoria2e5** [comment]
I already got rid of them

**Comment by Jack-Works** [comment]
Uhh, I found this not working on my machine.

WebCryptoLiner and WebExtension polyfill loaded, no logs printed, I think background service does not started

**Comment by Artoria2e5** [comment]
screenshot the dom tree will ya

**Comment by Jack-Works** [comment]
Nothing new.

```html
<html><head>
<script src="/polyfill/webcrypto-liner.shim.js"></script>
<script src="/polyfill/browser-polyfill.min.js"></script>
<script src="/js/vendors~app~backgroundservice~contentscript.chunk.js"></script><!-- XXX: bit hacky, relies on entry point list, need to read up on webpack plugins -->
<script src="/js/backgroundservice.js"></script>
</head><body>
</body></html>
```

**Comment by Artoria2e5** [comment]
i have no problem with the background, but the content script understandably fails to inject correctly the dependencies. gonna write a plugin for browser.geturl wrapping.

File: config-overrides.js [hunk 1]
```diff
@@ -23,25 +24,41 @@ module.exports = function override(/** @type{import("webpack").Configuration} */
...
+        }),
+        new HtmlWebpackPlugin({
+            chunks: ['options-page'],
+            filename: 'index.html',
+            ...includeWebExtPolyfill,
+        }),
+        new HtmlWebpackPlugin({
+            chunks: ['content-script'],
+            filename: 'generated__content__script.html',
+            ...includeWebExtPolyfill,
```
Comment (Misaka-0x447f) [inline config-overrides.js]: Consider make a factory for these duplicated code?

File: config-overrides.js [hunk 2]
```diff
@@ -10,9 +11,9 @@ module.exports = function override(/** @type{import("webpack").Configuration} */
...
     config.entry = {
         devtools: 'react-devtools',
-        app: path.join(__dirname, './src/index.tsx'),
-        contentscript: path.join(__dirname, './src/content-script.ts'),
-        backgroundservice: path.join(__dirname, './src/background-service.ts'),
+        'options-page': path.join(__dirname, './src/index.tsx'),
+        'content-script': path.join(__dirname, './src/content-script.ts'),
+        'background-service': path.join(__dirname, './src/background-service.ts'),
         injectedscript: path.join(__dirname, './src/extension/injected-script/index.ts'),
         qrcode: path.join(__dirname, './src/web-workers/QRCode.ts'),
```
Comment (Misaka-0x447f) [inline config-overrides.js]: Consider make a factory for this:
```javascript
(file) => path.join(__dirname, file)
```

**Pull request merged**